### PR TITLE
VScript changes

### LINKED ISSUE #29: VScript debug draw-related functions not functional
This was reported by NvC_DmN_CH on Mapbase's SMC channel.

This could be related to the fact these functions are located in `NDebugOverlay`, a separate namespace.

### PR DESCRIPTION
**Pull request by samisalreadytaken**
VScript changes that reflect additional utility and consistency with Source 2, L4D2 and CSGO.

CBaseEntity::SetThinkFunction sets the "ScriptThink" context (script func defined in entity thinkfunction kv), CBaseEntity::SetThink sets the "ScriptThinkH" context. SetThink stores and executes the handle directly, thus allows closure inputs.

On this matter, I suggest doing what Source 2 does, and remove the 'thinkfunction' keyvalue and the CBaseEntity::ScriptThink function (by extension SetThinkFunction too), and let the user handle their think functions more freely. Having it as a keyvalue forces a recompile if the function name is to change; but using the new SetThink function, they can just do it all in the script file `self.SetThink( function(){}, 0 )`

CBaseEntity::SetContextThink is also a possible addition for the future.

CBaseEntity::SetThinkFunction includes initial delay and removes the need for AddThinkToEnt. I suggest removal of the global AddThinkToEnt.

CDebugOverlayScriptHelper resolves #29. The reason NDebugOverlay code is copied here instead of calling it is because of the player and LOS checks. Rather than to change the functionality of the original code, I copied the important parts.

There is no need for metamethod stubs (excluding _tostring and IsValid). They will fail in all conditions except when the behaviour is manually defined, then there would already be no need for stubs.

The incorrect GetLeftVector naming was fixed, and GetPlayerUserId in L4D2 was also changed to its native name GetUserID in Source 2.

I'm not sure what the purpose of SetRenderColorR,G,B are (they're not even used in game code), they seem unnecessary when there is SetRenderColor with R,G,B params, also as seen in Source 2.

There is a SetOwner duplicate in baseentity (in Valve's code): ScriptSetOwner and (virtual) SetScriptOwnerEntity. Neither have any references other than script declarations. They could be cleaned up.

### CODE REVIEW

**Comment by Blixibon** [comment]
I love these changes and I would normally really want to merge this PR, but it currently breaks too much existing functionality to be acceptable into Mapbase as-is.

* Consistency with other Source (2) games is normally welcome, but I know some scripts are already using the functions renamed in this PR and it may be too late to change them for this. It's possible that all of the existing scripts are in mods with their own versions of Mapbase and if the involved mods were updated to a version of Mapbase which uses these new names, those scripts could presumably be updated as well. This may require further discussion.

* The metamethods removed in this PR are supposed to allow programmers to add member variables and calculations for script classes, like `Quaternion`. I don't really understand the reasoning for why they were removed since they're supposed to do nothing by default; Script classes are supposed to override them when they need to use them. How would there be no need for stubs if they're manually defined?

Maintaining legacy support is something I neglected to mention in the contribution guidelines, so I'll add a line for that in the next update.

Unless we decide to keep these renames in, I strongly recommend adding another commit which restores `GetPlayerUserID`, `GetLeftVector`, `GetTeamNumber`, etc. as deprecated names and also restores the removed metamethods. I could do this myself as well if you don't have time.

*(I will, however, unconditionally accept turning `IsServerScript()` and `IsClientScript()` into their shortened versions because I know nobody would've been using them yet)*

---

> I'm not sure what the purpose of SetRenderColorR,G,B are (they're not even used in game code), they seem unnecessary when there is SetRenderColor with R,G,B params, also as seen in Source 2.

They're shortcuts for when you want to set one color channel, but not the other and you don't want to type out something like `self.SetRenderColor( 255, self.GetRenderColorG(), self.GetRenderColorB() )` or involve a separate vector.

**Comment by samisalreadytaken** [comment]
After testing the metamethods a bit more, I see you have a different design than what I had in my mind, reverted.

You can revert the renamings as you see fit - in the end it is your decision. There is only 5, it shouldn't take more than 2 minutes anyway. The wiki could also only include the new names to discourage deprecated name usage.

File: sp/src/vscript/squirrel/include/sqstdtime.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
+//-----------------------------------------------------------------------
```
Comment (ReDucTor) [inline sp/src/vscript/squirrel/include/sqstdtime.h:1]: Should ideally keep squirrel as being unmodified

Comment (samisalreadytaken) [reply sp/src/vscript/squirrel/include/sqstdtime.h:1]: Modifying how, do you think the file should be located elsewhere?

Comment (ReDucTor) [reply sp/src/vscript/squirrel/include/sqstdtime.h:1]: This file is stored in the squirrel folder which is currently just a direct copy of squirrel with no changes, adding this file in that folder changea that

**Pull request merged**